core/src/gl/renderState.h

```diff
@@ -112,6 +113,9 @@ class RenderState {
 
     JobQueue jobQueue;
 
+    std::map<std::string, GLuint> fragmentShaders;
+    std::map<std::string, GLuint> vertexShaders;
```

Comment:
How about `unordered_map` for these? We don't need a specific ordering and it seems like hash tables would require fewer string comparisons.